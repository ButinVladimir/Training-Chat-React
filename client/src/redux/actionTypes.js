export const CHANGE_LOGIN = 'CHANGE_LOGIN';
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';
export const CHANGE_SPEECH = 'CHANGE_SPEECH';
export const SAY = 'SAY';
export const ADD_TO = 'ADD_TO';
export const REMOVE_TO = 'REMOVE_TO';
export const CONNECT = 'CONNECT';
export const DISCONNECT = 'DISCONNECT';
export const SHOW_ERROR = 'SHOW_ERROR';
export const HIDE_ERROR = 'HIDE_ERROR';
export const USER_APPEARS = 'USER_APPEARS';
export const USER_DISAPPEARS = 'USER_DISAPPEARS';
export const GET_MESSAGE = 'GET_MESSAGE';
export const UPDATE_USER_LIST = 'UPDATE_USER_LIST';
