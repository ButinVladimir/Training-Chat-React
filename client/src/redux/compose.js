export default function compose(...args) {
  return (state, action) => {
    let resultState = state;
    args.forEach((reducer) => { resultState = reducer(resultState, action); });

    return resultState;
  };
}
