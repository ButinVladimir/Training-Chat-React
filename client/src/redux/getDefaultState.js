export default () => ({
  loginPanel: {
    loginned: false,
    login: '',
  },
  usersPanel: {
    usersList: [],
  },
  speechPanel: {
    speech: '',
    to: [],
  },
  chatPanel: {
    messages: [],
  },
  connected: false,
  errorPanel: {
    showError: false,
    error: '',
  },
});
