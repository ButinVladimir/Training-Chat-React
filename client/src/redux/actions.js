import * as actionTypes from './actionTypes';

export const changeLogin = loginValue => ({
  type: actionTypes.CHANGE_LOGIN,
  loginValue,
});

export const login = () => ({
  type: actionTypes.LOGIN,
});

export const logout = () => ({
  type: actionTypes.LOGOUT,
});

export const changeSpeech = speechValue => ({
  type: actionTypes.CHANGE_SPEECH,
  speechValue,
});

export const say = (id, date) => ({
  type: actionTypes.SAY,
  id,
  date,
});

export const addTo = recipient => ({
  type: actionTypes.ADD_TO,
  recipient,
});

export const removeTo = recipient => ({
  type: actionTypes.REMOVE_TO,
  recipient,
});
