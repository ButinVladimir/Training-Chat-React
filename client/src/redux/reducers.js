import compose from './compose';
import loginPanel from './reducers/loginPanel';
import speechPanel from './reducers/speechPanel';

const reducers = compose(loginPanel, speechPanel);

export default (state = {
  loginPanel: {
    loginned: false,
    login: '',
  },
  usersPanel: {
    usersList: [],
  },
  speechPanel: {
    speech: '',
    to: [],
  },
  chatPanel: {
    messages: [],
  },
}, action) => reducers(state, action);
