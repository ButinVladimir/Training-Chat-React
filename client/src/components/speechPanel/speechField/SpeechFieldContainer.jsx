import { connect } from 'react-redux';
import SpeechField from './SpeechField';
import { changeSpeech, say } from '../../../redux/actions';

const mapStateToProps = state => ({
  speech: state.speechPanel.speech,
  to: state.speechPanel.to,
});

const mapDispatchToProps = dispatch => ({
  onChange: speechValue => dispatch(changeSpeech(speechValue)),
  onSay: () => dispatch(say(`${Math.random() * Math.PI}-${Date.now()}`, (new Date(Date.now()).toLocaleString()))),
});

const SpeechFieldContainer = connect(mapStateToProps, mapDispatchToProps)(SpeechField);

export default SpeechFieldContainer;
