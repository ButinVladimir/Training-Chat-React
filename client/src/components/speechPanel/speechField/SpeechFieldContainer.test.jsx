import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import SpeechFieldContainer from './SpeechFieldContainer';
import { changeSpeech } from '../../../redux/actions';
import { SAY } from '../../../redux/actionTypes';

describe('SpeechFieldContainer', () => {
  let mockStore;

  beforeEach(() => {
    mockStore = configureStore();
  });

  it('renders without crashing', () => {
    const store = mockStore({
      speechPanel: {
        speech: 'speechValue',
        to: ['to1', 'to2', 'to3'],
      },
    });
    const component = mount(
      <Provider store={store}>
        <SpeechFieldContainer />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });

  it('handles changing speech', () => {
    const store = mockStore({
      speechPanel: {
        speech: '',
        to: [],
      },
    });
    const component = mount(
      <Provider store={store}>
        <SpeechFieldContainer />
      </Provider>,
    );

    const newToValue = 'newToValue';
    component.find('input.speech').simulate('change', { target: { value: newToValue } });

    const actions = store.getActions();
    const expectedActions = [changeSpeech(newToValue)];
    expect(actions).toMatchObject(expectedActions);
  });

  it('handles saying', () => {
    const store = mockStore({
      speechPanel: {
        speech: 'speechValue',
        to: [],
      },
    });
    const component = mount(
      <Provider store={store}>
        <SpeechFieldContainer />
      </Provider>,
    );

    component.find('button.say').simulate('click');

    const actions = store.getActions();
    const expectedActions = [{ type: SAY }];
    expect(actions).toMatchObject(expectedActions);
  });
});
