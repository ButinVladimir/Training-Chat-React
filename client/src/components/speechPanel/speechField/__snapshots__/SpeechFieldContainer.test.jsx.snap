// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeechFieldContainer renders without crashing 1`] = `
<div
  className="speech-container"
>
  <div
    className="speech-input-container"
  >
    <input
      className="speech"
      onChange={[Function]}
      type="text"
      value="speechValue"
    />
    <button
      className="say"
      onClick={[Function]}
      type="button"
    >
      Say
    </button>
  </div>
  <div
    className="speech-to-container"
  >
    <button
      className="to-button"
      onClick={[Function]}
      type="button"
    >
      @to1
    </button>
    <button
      className="to-button"
      onClick={[Function]}
      type="button"
    >
      @to2
    </button>
    <button
      className="to-button"
      onClick={[Function]}
      type="button"
    >
      @to3
    </button>
  </div>
</div>
`;
